management-api-server/src/main/java/com/datastax/mgmtapi/resources/common/BaseResources.java

```diff
@@ -71,4 +73,21 @@ protected Response handle(Callable<Response> action) {
           .build();
     }
   }
+
+  protected boolean keyspaceExists(String keyspaceName) {
+    if (keyspaceName != null) {
+      try {
+        ResultSet result =
+            app.cqlService.executePreparedStatement(
+                app.dbUnixSocketFile, "CALL NodeOps.getKeyspaces()");
+        Row row = result.one();
+        if (row != null) {
+          return row.getList(0, String.class).contains(keyspaceName);
+        }
+      } catch (Exception e) {
+        // cxonnection issue or some other problem. Eat it and return false.
```

Comment:
Suggestion: Can we throw an exception here so that we can ultimately return a 500 error? I think false is a bit different to a genuine error, even though in this use case the difference is a bit subtle. One is (as you currently have) a 404, the other is a 500.

Reply:
We can. I initially just wanted this as a utility method that only returned `true` if the keyspace exists, and false otherwise. But we could leverage this in a future V2 `getKeyspace` endpoint. Right now, the `listKeyspaces` endpoint simply returns an empty list if the keyspace specified does not exist.

Reply:
I don't think this suggestion has been implemented, is it something we want to go for?

Reply:
Ignore me, it has been implemented.